packages/did-provider-ebsi/src/ebsi-did-onboarding.ts

```diff
@@ -0,0 +1,309 @@
+import {
+  createVerifiablePresentationJwt,
+  EbsiIssuer,
+  EbsiVerifiablePresentation,
+} from '@cef-ebsi/verifiable-presentation'
+import { IIdentifier } from '@veramo/core'
+import * as jose from 'jose'
+import {
+  IKeyJwks,
+  IRPCResult,
+  ISession,
+  IVerifiableAuthorization,
+  IVerifiablePresentation,
+} from './types/ebsi-provider-types'
+import { v4 as uuidv4 } from 'uuid'
+import { randomBytes } from 'crypto'
+import { ethers } from 'ethers'
+import { Agent } from '@cef-ebsi/siop-auth'
+import { privateKeyJwkToHex } from './ebsi-did-utils'
+
+// Main onboard function to call when creating a new identifier
+export async function onboard(args: {
+  bearer: string
+  keyJwks: IKeyJwks
+  identifier: Omit<IIdentifier, 'provider'>
+}): Promise<any> {
+  if (!args.identifier.controllerKeyId) {
+    throw new Error('Missing controllerKeyId')
+  }
+  const kid = args.identifier.controllerKeyId
+  const subject = kid.split('#')[1]
+  const idToken = {
+    sub: subject,
+    sub_jwk: args.keyJwks.publicKeyJwk,
+    nonce: uuidv4(),
+    responseMode: 'form_post',
+  }
+
+  const privateKey = await jose.importJWK(args.keyJwks.privateKeyJwk, 'ES256K')
+  const idTokenJwt = await new jose.SignJWT(idToken)
+    .setProtectedHeader({ alg: 'ES256K', typ: 'JWT', kid })
+    .setIssuedAt()
+    .setAudience('https://api-pilot.ebsi.eu/users-onboarding/v2/authentication-responses')
```

Comment:
Move string constants to separate file or something like that.

Split domain and path, e.g., DOMAIN=https://api-pilot.ebsi.eu and AUTH_RESP=${DOMAIN}/... .

There are multiple EBSI envs, so you can just replace domain and it works for other envs as well.

Reply:
Repeat everywhere